### DISCUSSION TITLE
Excalidraw as a extension

### DISCUSSION THREAD

**Discussion by lkuoch** [toeverything/AFFiNE#5165] [Ideas]
I really love drawing and editing diagrams on excalidraw. Their user experience is truly second to none.

Affine has something special with the combined note + whiteboard experience.

To elevate it to the next level I would love to see Excalidraw supported seamlessly as a plugin or integrated natively

**Comment by EYHN** [comment]
In the future, we will support inserting external content by plugins, such as figma, google map, etc. excalidraw should also be able to do so.

In addition, we will work hard to enhance affine’s whiteboard function and experience to reach the first-class level.

**Reply by jsCarvalhoNeto** [reply]
I love the whiteboard the way it is, I just need a way to display the slide transition in a smoother/fluid way like it does in excalidraw/obsidian.

**Reply by Pe4enuga** [reply]
true. I love using default whiteboard, but I nead to miggrate my data from another services

**Comment by HansAndreManfredson** [comment]
Any updates?

**Comment by HansAndreManfredson** [comment]
Any updates?

**Comment by MrAndrewMal** [comment]
Any updates?

**Comment by CMCDragonkai** [comment]
Perhaps one can "embed" excalidraw drawings. But they rely on links. Excalidraw objects themselves would be nice to support instead, although it does support export as SVGs.

**Comment by UnoYakshi** [comment]
AFFiNE would definitely win from integrating interactive Excalidraw boards.

**Comment by spmp** [comment]
I have extended AFFiNES drawing capability to be on-par (at a base level) with Drawio. This may meet your requirement. Preparing the PR for it today.

May need support as this is a crazy big chunk of change. Testers please!

See Discussion at  https://github.com/toeverything/AFFiNE/discussions/14471 and PR #14720 
<img width="1087" height="830" alt="Screenshot From 2026-02-26 08-55-54" src="https://github.com/user-attachments/assets/fcc5c015-746c-4e4d-aeca-4cd7a96b2911" />
